programs/survival/portalorient.sc

```diff
@@ -0,0 +1,186 @@
+__config() ->
+(
+	m(
+		l('scope','player'),
+		l('stay_loaded',true),
+		
+		l('commands',
+			m(
+				l('<mode>','__change')
+			)
+		),
+		l('arguments',
+			m(
+				l('mode',
+					m(
+						l('type','term'),
+						l('options',
+							l('off','air','solid')
+						),
+						l('suggest',
+							l('off','air','solid')
+						)
+					)
+				)
+			)
+		)
+	)
+);
+
+//setup the script's data if nothing is there already
+//it's an empty map, so players without data will have a null mode
+__on_start() ->
+(
+	if(!load_app_data(),
+		store_app_data(m());
+	)

```

Comment:
Just for the sake of curiosity, hw hard would it be to automatically make the players be in off mode instead of saving an empty map? (Im genuinely asking cos I have no clue how tf store_app_data works and whether it will overwrite other player's data or otherwise, idk)

Reply:
It should only write the empty map if there's no app data already stored, so what gets overwritten is just `null`; no player data can be overwritten(in theory, but I can't think of any edge cases).

Reply:
I suppose `false` can also get overwritten, but either way a map with non-null keys and values should pass the check and `!load_app_data()` should return `false`, preventing `store_app_data()` from getting called.

Reply:
Ah ok